moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -1941,37 +1938,47 @@ double moveit::core::RobotState::computeCartesianPath(const JointModelGroup* gro
     if (setFromIK(group, pose, link->getName(), 1, 0.0, validCallback, options))
     {
       traj.push_back(RobotStatePtr(new RobotState(*this)));
-
-      // compute the distance to the previous point (infinity norm)
-      if (test_joint_space_jump)
-      {
-        double dist_prev_point = traj.back()->distance(*traj[traj.size() - 2], group);
-        dist_vector.push_back(dist_prev_point);
-        total_dist += dist_prev_point;
-      }
     }
     else
       break;
     last_valid_percentage = percentage;
   }
 
-  if (test_joint_space_jump && !dist_vector.empty())
+  if (test_joint_space_jump)
   {
-    // compute the average distance between the states we looked at
-    double thres = jump_threshold * (total_dist / (double)dist_vector.size());
-    for (std::size_t i = 0; i < dist_vector.size(); ++i)
-      if (dist_vector[i] > thres)
-      {
-        logDebug("Truncating Cartesian path due to detected jump in joint-space distance");
-        last_valid_percentage = (double)i / (double)steps;
-        traj.resize(i);
-        break;
-      }
+    last_valid_percentage *= testJointSpaceJump(group, traj, jump_threshold);
   }
 
   return last_valid_percentage;
 }
 
+double moveit::core::RobotState::testJointSpaceJump(const JointModelGroup* group, std::vector<RobotStatePtr>& traj,
+                                                    double jump_threshold)
+{
+  std::vector<double> dist_vector;
+  double total_dist = 0.0;
+  for (std::size_t i = 1; i < traj.size(); ++i)
+  {
+    double dist_prev_point = traj[i]->distance(*traj[i - 1], group);
+    dist_vector.push_back(dist_prev_point);
```

Comment:
Now the final size of `dist_vector` is `(traj.size-1)*2` with a lot of zeros in the beginning...